hearth/lib/hearth/http/client.rb

```diff
@@ -128,8 +128,8 @@ def build_net_request(request)
       end
 
       # @param [Net::HTTP::Response] response
-      # @return [Hash<String, String>]
-      def extract_headers(response)
+      # @return [Hash<String, Field>]
```

Comment:
This isn't really a Field class at this point is it?  This is just going to be a string from net:http's response right?  I'd argue for keeping this as extract_headers, since thats what its doing.  `Field` is a concept on Hearth (smithy) responses, not on Net http responses